src/suricata.c

```diff
@@ -2663,6 +2639,11 @@ int PostConfLoadedSetup(SCInstance *suri)
 
     MacSetRegisterFlowStorage();
 
+    LiveDeviceFinalize(); // must be after EBPF extension registration
```

Comment:
this now runs before `ConfigGetCaptureValue` which relies on the livedevs freed by ` LiveDeviceFinalize`:
```
Starting program: /home/victor/dev/suricata/src/suricata -vvvvv -c /home/victor/dev/qa/ips/tests/geneve//suricata.yaml --af-packet -S /home/victor/dev/qa/ips/tests/geneve//test.rules -l /tmp//geneve-decoder/suricata/
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Notice: suricata: This is Suricata version 7.0.0-rc1-dev (b754887e6 2023-01-17) running in SYSTEM mode [suricata.c:1138]
Info: cpu: CPUs/cores online: 8 [util-cpu.c:182]
Info: af-packet: Setting IPS mode [runmode-af-packet.c:150]

Program received signal SIGSEGV, Segmentation fault.
strlcpy (dst=0x7fffffffe060 " \271\002\021\001\006#", src=0x316634736e65 <error: Cannot access memory at address 0x316634736e65>, siz=128) at util-strlcpyu.c:55
55                  if ((*d++ = *s++) == 0)
(gdb) bt
#0  strlcpy (dst=0x7fffffffe060 " \271\002\021\001\006#", src=0x316634736e65 <error: Cannot access memory at address 0x316634736e65>, siz=128) at util-strlcpyu.c:55
#1  0x0000555555cc510f in ConfigGetCaptureValue (suri=0x5555575b1360 <suricata>) at suricata.c:2435
#2  0x0000555555cc59c9 in PostConfLoadedSetup (suri=0x5555575b1360 <suricata>) at suricata.c:2660
#3  0x0000555555cc65b6 in SuricataMain (argc=9, argv=0x7fffffffe2b8) at suricata.c:2901
#4  0x0000555555cbffdd in main (argc=9, argv=0x7fffffffe2b8) at main.c:22
```

Reply:
Ah thanks, didn't catch that without the sanitizer enabled. 
Removed functions that work with pre-registered devices and used functions that work with registered devices only (as we are after `LiveDeviceFinalize` function).